File: C++/Digits/program13.cpp
//Accept number from user and display addition of its even digits

#include <iostream>
using namespace std;

class Number
{
    public:
        int iNo;

        void Accept() // Setter
        {
            cout << "Enter the number" << endl;
            cin >> iNo;
        }

        void Display() // Getter
        {
            cout << "The number is: " << this->iNo << endl;
        }

        int SumEven()
        {
            int iDigit = 0;
            int iSum = 0;

            if(iNo < 0)
            {
                iNo = -iNo;
            }

            while(iNo > 0)
            {
                iDigit = iNo % 10;
                if((iDigit % 2) ==  0)
                {
                    iSum = iSum + iDigit;
                }
                iNo = iNo /10;
            }
            return iSum;
        }
};

int main()
{
    int iRet = 0;

    Number nobj;

    nobj.Accept();
    nobj.Display();

    iRet = nobj.SumEven();
    cout<<"Addition of even digits are : "<<iRet<<endl;
    
    return 0;
}

File: Java/Digit/program13.java
//Accept number from user and display addition of its even digits

import java.util.*;

class Digit
{
    public int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the number");
        iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("The number is: "+this.iNo);
    }

    public int SumEven()
    {
        int iDigit = 0;
        int iSum = 0;

        if(iNo < 0)
        {
            iNo = -iNo;
        }

        while(iNo > 0)
        {
            iDigit = iNo % 10;
            if((iDigit % 2) ==  0)
            {
                iSum = iSum + iDigit;
            }
            iNo = iNo /10;
        }
        return iSum;
    }
}

class program13
{
    public static void main(String arg[])
    {
        int iRet = 0;

        Digit obj = new Digit();

        obj.Accept();
        obj.Display();

        iRet = obj.SumEven();
        System.out.println("Addition of even digits are : "+iRet);
    }
}